bin/ruby-build

```diff
@@ -300,42 +300,40 @@ http() {
   local file="$3"
   [ -n "$url" ] || return 1
 
-  if type curl &>/dev/null; then
+  if type aria2c &>/dev/null; then
+    "http_${method}_aria2c" "$url" "$file"
+  elif type curl &>/dev/null; then
     "http_${method}_curl" "$url" "$file"
   elif type wget &>/dev/null; then
     "http_${method}_wget" "$url" "$file"
   else
-    echo "error: please install \`curl\` or \`wget\` and try again" >&2
+    echo "error: please install \`aria2c\`, \`curl\` or \`wget\` and try again" >&2
     exit 1
   fi
 }
 
+http_head_aria2c() {
+  aria2c --dry-run ${ARIA2_OPTS} "$1" >&4 2>&1
```

Comment:
Yes. In aria2, it seems "dry run" means "not perfirming any GET requests". I confirmed it with tcpdump.